packages/astro/src/core/errors/utils.ts

```diff
@@ -0,0 +1,303 @@
+import eol from 'eol';
+import { BuildResult } from 'esbuild';
+import * as fs from 'fs';
```

Comment:
notice that this file heavily depends on Node.js. If the error types are going to be used in runtime/ or in core/render (not core/render/dev) then that's going to be an issue since those parts are included in SSR so they have to be platform independent.

This is why core/render has the dev/ subfolder, that's why vite/dev specific stuff can go. Don't love the pattern or anything but that's how we handle it for now. Might likely need some similar split so that the runtime's errors can be imported without bringing in Node deps.